cpp/src/binaryop/binaryop.cpp

```diff
@@ -75,7 +75,7 @@ std::pair<rmm::device_buffer, size_type> scalar_col_valid_mask_and(
  * @brief Does the binop need to know if an operand is null/invalid to perform special
  * processing?
  */
-inline bool is_null_dependent(binary_operator op)
+inline static bool is_null_dependent(binary_operator op)
```

Comment:
This is really interesting given https://clang.llvm.org/extra/clang-tidy/checks/misc/use-anonymous-namespace.html :D
I guess that one does not have an automatic fix :)

Reply:
I'd prefer anonymous namespace instead of static. Seems more modern?

Reply:
It's unfortunate that https://clang.llvm.org/extra/clang-tidy/checks/misc/use-internal-linkage.html has a fix available but https://clang.llvm.org/extra/clang-tidy/checks/misc/use-anonymous-namespace.html does not. clang-tidy applies the former autofix but then complains about the latter. I think we still want both enabled though because it doesn't detect functions that aren't in an anonymous namespace or static but should be with just the second rule. The second rule simply sees the static keyword on functions and then checks if an anonymous namespace would be appropriate.

Reply:
I think one issue may be that since we are omitting cu files clang-tidy considers some functions as only needing internal visibility because it's not seeing the usage in other TUs that are compiled by nvcc (cu files).